### PR TITLE
[error] move the debug assertion properly so it doesn't panic under O…

### PR DESCRIPTION
**Pull request by zekun000**
…UT_OF_GAS

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

### CODE REVIEW

**Comment by vgao1996** [comment]
@tnowacki I remember you trying a similar fix a long time ago... Any idea what prevented that from happening?

**Comment by tnowacki** [comment]
This seems like a reasonable change. 
And @vgao1996, Bob was going to take a look but never got around to it IIRC

File: language/move-binary-format/src/errors.rs [hunk 1]
```diff
@@ -82,10 +82,17 @@ impl VMError {
...
                 if major_status.status_type() == StatusType::Execution =>
             {
+                let abort_location = match &location {
+                    Location::Script => vm_status::AbortLocation::Script,
+                    Location::Module(id) => vm_status::AbortLocation::Module(id.clone()),
+                    Location::Undefined => {
+                        return VMStatus::Error(major_status);
+                    }
+                };
+                // Errors for OUT_OF_GAS do not always have index set, it should already return above.
```
Comment (tnowacki) [inline language/move-binary-format/src/errors.rs:95]: This should read a bit tighter than that
```suggestion
                // Errors for OUT_OF_GAS do not always have index set: if it does not, it should already return above.
```

**Review by tnowacki — APPROVED** [review]

**Comment by vgao1996** [comment]
@zekun000 we don't have bors in this repo any more... You simply need to make sure all your tests pass and then there will be a rebase button you can push. (In this particular case, it's likely a flaky test. I've issued a re-run for you)

**Pull request merged**